src/Lecture2.hs

```diff
@@ -191,7 +235,19 @@ verify that.
 [1,2,3,4,7]
 -}
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge x1 x2 = go [] x1 x2
+    where
+        go :: [Int] -> [Int] -> [Int] -> [Int]
+        go list [] [] = list
+        go list [x] [] = go (list ++ [x]) [] []
+        go list [] [y] = go (list ++ [y]) [] []
+        go list x [] = go (list ++ x) [] []
+        go list [] y = go (list ++ y) [] []
+        go list (x:xs) (y:ys) = 
+            if x <= y 
+            then go (list++[x])  xs (y:ys) 
+            else go (list++[y]) (x:xs) ys
```

Comment:
It is more efficient to construct lists using the cons operator (`:`) directly without introducing intermediate values or adding lists with `(++)` whenever possible. Adding elements to the end of the list is rather slow also 🐌